### DISCUSSION TITLE
pb-import:  CSV, TSV, SQLite DB data imports

### DISCUSSION THREAD

**Discussion by burggraf** [pocketbase/pocketbase#4108] [Show and tell]
[pb-import](https://github.com/burggraf/pb-import)

So far, I've done SQLite import only.  This tool will:

1. read your external SQLite database
2. get a list of the tables
3. read the column names and types from each table
4. create a collection based on those fields
5. import the data from the external database into Pocketbase
6. continue on with all the other tables
7. you can specify just one a custom set of tables to be imported (if you don't want the entire database)

**Comment by burggraf** [comment]
CSV, TSV, (any delimited) file imports are now supported.  This tool will:

1. parse your entire file (CSV, TSV, any type of delimited file with any standard row and column delimiters)
2. determine data types for your fields
3. create your new collection
4. load data into your new collection

**Comment by tigawanna** [comment]
Consider listing it in awesome pocketbase

**Comment by bsmithuk** [comment]
I've tried importing an sqlite database I  generated from another non-database application (BlenderBIM - As interested in looking at this as a tool). The script runs fine but each creating collection fails:
024/06/04 21:36:24 *** IfcZShapeProfileDef
2024/06/04 21:36:24 *********************
2024/06/04 21:36:24 creating collection IfcZShapeProfileDef...
2024/06/04 21:36:24 create error:  GoError: SQL logic error: no such table: _collections (1) 

I've tried opening the sqlite in PowerBi and online viewers and the tables exist. Any ideas of how to resolve?
